app/nemeton/store.go

```diff
@@ -433,3 +433,127 @@ func (s *Store) GetPhaseBlocks(ctx context.Context, number int) (*BlockRange, er
+	res := s.db.Collection(phasesCollectionName).
+		FindOne(ctx, bson.M{"blocks.to": height}, options.FindOne())
+	if err := res.Err(); err != nil {
+		if errors.Is(err, mongo.ErrNoDocuments) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	var phase Phase
+	return &phase, res.Decode(&phase)
+}
+
+// CompleteValidatorsUptimeForPhase is used to concat all missed blocks on a given phase and calculate the number of
+// points rewarded.
+//
+//nolint:funlen
+func (s *Store) CompleteValidatorsUptimeForPhase(ctx context.Context, phase *Phase) error {
+	var taskID string
+	if phase != nil {
+		for _, task := range phase.Tasks {
+			if task.Type == taskTypeUptime {
+				taskID = task.ID
+				break
+			}
+		}
+	}
+
+	_, err := s.db.Collection(validatorsCollectionName).Aggregate(ctx, bson.A{
+		bson.M{"$lookup": bson.M{
+			"from": "phases",
+			"pipeline": bson.A{
+				bson.M{"$match": bson.M{"_id": phase.Number}},
+				bson.M{"$project": bson.M{"blocks": 1}},
+			},
+			"as": "currentPhase",
+		}},
+		bson.M{"$unwind": bson.M{"path": "$missedBlocks", "preserveNullAndEmptyArrays": true}},
+		bson.M{"$unwind": bson.M{"path": "$currentPhase"}},
+		bson.M{"$replaceRoot": bson.M{
+			"newRoot": bson.M{
+				"count": bson.M{
+					"$switch": bson.M{
+						"branches": bson.A{
+							bson.M{
+								"case": bson.M{"$and": bson.A{
+									bson.M{"$gte": bson.A{"$missedBlocks.from", "$currentPhase.blocks.from"}},
+									bson.M{"$lte": bson.A{"$missedBlocks.to", "$currentPhase.blocks.to"}},
+								}},
+								"then": bson.M{"$subtract": bson.A{"$missedBlocks.to", "$missedBlocks.from"}},
+							},
+							bson.M{
+								"case": bson.M{"$and": bson.A{
+									bson.M{"$lt": bson.A{"$missedBlocks.from", "$currentPhase.blocks.from"}},
+									bson.M{"$lte": bson.A{"$missedBlocks.to", "$currentPhase.blocks.to"}},
+									bson.M{"$gte": bson.A{"$missedBlocks.to", "$currentPhase.blocks.from"}},
+								}},
+								"then": bson.M{"$subtract": bson.A{"$missedBlocks.to", "$currentPhase.blocks.from"}},
+							},
+							bson.M{
+								"case": bson.M{"$and": bson.A{
+									bson.M{"$gte": bson.A{"$missedBlocks.from", "$currentPhase.blocks.from"}},
+									bson.M{"$gt": bson.A{"$missedBlocks.to", "$currentPhase.blocks.to"}},
+									bson.M{"$lte": bson.A{"$missedBlocks.from", "$currentPhase.blocks.to"}},
+								}},
+								"then": bson.M{"$subtract": bson.A{"$currentPhase.blocks.to", "$missedBlocks.from"}},
+							},
+						},
+						"default": 0,
+					},
+				},
+				"currentPhase": "$currentPhase",
+				"validator":    "$_id",
+				"points":       "$points",
+			},
+		}},
+		bson.M{"$group": bson.M{
+			"_id":              "$validator",
+			"totalMissedBlock": bson.M{"$sum": "$count"},
+			"phase":            bson.M{"$mergeObjects": "$currentPhase"},
+			"points":           bson.M{"$first": "$points"},
+		}},
+		bson.M{"$addFields": bson.M{
+			"uptime": bson.M{"$subtract": bson.A{
+				bson.M{
+					"$pow": bson.A{2501, bson.M{
```

Comment:
The uptime won't attribute the same number of points every phase, parameters have been introduced in the task structure so you can retrieve this value with: `task.GetUptimeMaxPoints() + 1`

Reply:
Ah yes ! I've seen that has been introduce in the last PR #43, I will update this.